### ISSUE TITLE
Split/modularize it?

### ISSUE THREAD

**Issue by rugk** [Pernosco/DOMRec#1]
I guess having one single JS file with > 1000 LOC is not really good.
It would likely be good to split it into multiple files or so and possibly use WebPack or so to compile it.

**Comment by bendlas** [comment]
Doesn't that go counter to the supposed goal of pasting the code to browser console?

**Comment by rugk** [comment]
Ah that was an idea?

But no, it does not. At the end of the build process you anyway only get one file.
You might say you trust the source code more, but if it is so long, is that really an argument?
Also don't know whether pasting things into the browser is really the proper way. IMHO a browser extension would be much cleaner then. (A bookmarklet may also work, though, but there it cannot be split either/respectively only the compiled version can be saved.)

**Comment by casaper** [comment]
I would be willing to contribute to port that to TypeScript. The finalized transpiled file would still be one file.

I really like the idea of DOMRec and see some potential behind the concept that goes far beyond just recording a personal web session. Especially in CI testing and Snapshot tests I see this as something that could, at least in theory, be very usefull.

**Comment by sopelt** [comment]
Yes, recording E2E-Tests could be really nice!

**Comment by casaper** [comment]
@rugk I just now started a draft PR regarding this now. https://github.com/Pernosco/DOMRec/pull/3

This is in super early stage though. For now I just did so much that tsc will not throw when transpiling.

It is very much WIP and once it is finished it will only be a first step still.

**Comment by rugk** [comment]
Oh that's great! Also TypeScript is a good idea.

**Comment by Junkern** [comment]
I also have done some work on this (https://github.com/Junkern/DOMRec). I have also ported it to Typescript and used webpack. There are two files now: one for recording and one for replaying.
